Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -872,22 +874,25 @@ inline RefPtr<StylePathData> BuilderConverter::convertDPath(BuilderState& builde
 inline RefPtr<PathOperation> BuilderConverter::convertPathOperation(BuilderState& builderState, const CSSValue& value)
 {
     if (auto* primitiveValue = dynamicDowncast<CSSPrimitiveValue>(value)) {
-        if (primitiveValue->isURI()) {
-            auto cssURLValue = primitiveValue->stringValue();
-            auto fragment = SVGURIReference::fragmentIdentifierFromIRIString(cssURLValue, builderState.document());
-            // FIXME: It doesn't work with external SVG references (see https://bugs.webkit.org/show_bug.cgi?id=126133)
-            const TreeScope* treeScope = nullptr;
-            if (builderState.element())
-                treeScope = &builderState.element()->treeScopeForSVGReferences();
-            else
-                treeScope = &builderState.document();
-            auto target = SVGURIReference::targetElementFromIRIString(cssURLValue, *treeScope);
-            return ReferencePathOperation::create(cssURLValue, fragment, dynamicDowncast<SVGElement>(target.element.get()));
-        }
-        ASSERT(primitiveValue->valueID() == CSSValueNone);
+        ASSERT_UNUSED(primitiveValue, primitiveValue->valueID() == CSSValueNone);
         return nullptr;
     }
 
+    if (RefPtr url = dynamicDowncast<CSSURLValue>(value)) {
+        auto styleURL = Style::toStyle(url->url(), builderState);
+
+        auto fragment = SVGURIReference::fragmentIdentifierFromIRIString(styleURL, builderState.document());
+        // FIXME: It doesn't work with external SVG references (see https://bugs.webkit.org/show_bug.cgi?id=126133)
```

Comment:
```suggestion
        // FIXME: This doesn't work correctly for external SVG references (see https://bugs.webkit.org/show_bug.cgi?id=126133)
```
Also would love to be more specific than “doesn’t work correctly” if I knew what was wrong.

Reply:
This one I totally do know the answer and can improve that comment.